nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseSyslog5424.java

```diff
@@ -0,0 +1,174 @@
+    public static final AllowableValue NULL = new AllowableValue(NilPolicy.NULL.name(),NilPolicy.NULL.name(),"The missing field will have an empty attribute added.");
+    public static final AllowableValue DASH = new AllowableValue(NilPolicy.DASH.name(),NilPolicy.DASH.name(),"The missing field will have an attribute added with the value of '-'.");
+
+    public static final PropertyDescriptor CHARSET = new PropertyDescriptor.Builder()
+        .name("Character Set")
+        .description("Specifies which character set of the Syslog messages")
+        .required(true)
+        .defaultValue("UTF-8")
+        .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor NIL_POLICY = new PropertyDescriptor.Builder()
+        .name("nil_policy")
+        .displayName("NIL Policy")
+        .description("Defines how NIL values are handled for header fields.")
+        .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+        .allowableValues(OMIT,NULL,DASH)
+        .required(true)
+        .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+        .defaultValue(NULL.getValue())
+        .build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+        .name("failure")
+        .description("Any FlowFile that could not be parsed as a Syslog message will be transferred to this Relationship without any attributes being added")
+        .build();
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("Any FlowFile that is successfully parsed as a Syslog message will be to this Relationship.")
+        .build();
+
+    private volatile StrictSyslog5424Parser parser;
+
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>(2);
+        properties.add(CHARSET);
+        properties.add(NIL_POLICY);
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_SUCCESS);
+        return relationships;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String charsetName = context.getProperty(CHARSET).getValue();
+        final String nilPolicyString = context.getProperty(NIL_POLICY).getValue();
+
+        // If the parser already exists and uses the same charset, it does not need to be re-initialized
+        if (parser == null || !parser.getCharsetName().equals(charsetName)) {
```

Comment:
I thought I wrote a comment here yesterday, but it seems to have not made it through..

If the processor is configured with concurrent tasks > 1 then multiple threads may enter this logic concurrently and be in a race condition to initialize the parser. It may not be a big problem, but usually we try to protect against this.

Since the parser here is not making any external calls when being constructed, it would safe to construct it in an @OnSchedule method every time the processor starts.

If you prefer the lazy init in onTrigger then usually a check-synchronize-recheck pattern works... 
```

if (parser == null || !parser.getCharsetName().equals(charsetName)) {
  synchronized(this) {
      if (parser == null || !parser.getCharsetName().equals(charsetName)) {
          parser = new StrictSyslog5424Parser(Charset.forName(charsetName),NilPolicy.valueOf(nilPolicyString));
      }
  }
}
```

Reply:
done